app/javascript/packs/admin_evaluations/new_evaluation.jsx

```diff
@@ -0,0 +1,219 @@
+import React from 'react';
+import Select from 'react-normalized-select';
+import AlertContainer from 'react-alert';
+import { Link } from 'react-router-dom';
+import { Redirect } from 'react-router';
+import Fetch from '../Fetch';
+
+class NewAdminEvaluation extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+    this.userId = this.props.match.params.user_id;
+
+    this.emptyGoal = {
+      name: '',
+      mark: '',
+      id: null,
+      _destroy: false,
+    };
+
+    this.state = {
+      redirect: false,
+      goalName: '',
+      goalMark: '0',
+      evaluation: {
+        dueDate: '',
+        goals_attributes: [Object.assign({}, this.emptyGoal)]
+      },
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleDueDateChange = this.handleDueDateChange.bind(this);
+    this.handleGoalNameChange = this.handleGoalNameChange.bind(this);
+    this.handleMarkChange = this.handleMarkChange.bind(this);
+    this.handleAddGoal = this.handleAddGoal.bind(this);
+    this.handleRemoveGoal = this.handleRemoveGoal.bind(this);
+  }
+
+  componentWillMount() {
+    this.state.evaluation.goals_attributes[0]._destroy = true;
+  }
+
+  componentDidMount() {
+    $('#add-goal').hide();
+    $('#submit').hide();
+  }
+
+  handleSubmit() {
+    const dueDate = this.state.evaluation.dueDate;
+    const evaluationGoals = this.state.evaluation.goals_attributes;
+    Fetch.postJSON(`/api/v1/admin/user/users/${this.userId}/evaluations`, {
+      evaluation: {
+        user_id: this.userId,
+        due_date: dueDate,
+        goals_attributes: evaluationGoals,
+      },
+    })
+      .then(() => {
+        this.msg.success('Successfully created evaluation!');
+        this.setState({ redirect: true });
+      }).catch((errorResponse) => {
+        this.msg.error(errorResponse.errors);
+      });
+  }
+
+  handleDueDateChange(e) {
+    const evaluation = this.state.evaluation;
+    evaluation.dueDate = e.target.value;
+    this.setState({ evaluation: this.state.evaluation });
+  }
+
+  handleGoalNameChange(e) {
+    const goal = this.emptyGoal;
+    goal.name = e.target.value;
+    this.setState({ goalName: e.target.value });
+    if (goal.name.length > 0) {
```

Comment:
each time we are changing the state the component is rerendered. It doesn't make sense to do manipulations with DOM here.